lib/Sema/BuilderTransform.cpp

```diff
@@ -1485,9 +1485,11 @@ Optional<BraceStmt *> TypeChecker::applyFunctionBuilderBodyTransform(
   // If we encountered an error or there was an explicit result type,
   // bail out and report that to the caller.
   auto &ctx = func->getASTContext();
-  auto request = PreCheckFunctionBuilderRequest{AnyFunctionRef(func)};
-  switch (evaluateOrDefault(
-              ctx.evaluator, request, FunctionBuilderBodyPreCheck::Error)) {
+  auto request =
+      PreCheckFunctionBuilderRequest{{AnyFunctionRef(func),
+                                      /*SuppressDiagnostics=*/false}};
```

Comment:
Hmm, why is it necessary to do this in `applyFunctionBuilderBodyTransform` before calling `matchFunctionBuilder` which does `PreCheckFunctionBuilderRequest` again? Is this duplicated work, or is this doing something different than the one in `matchFunctionBuilder`?

Reply:
Yeah, this pre-check is diagnosing errors and returns where the one in matchFunctionBuilder just returns type match result.

Reply:
Ah, I understand. It's possible to diagnose early and bail out when the transform is applied to a function body, but not when it's applied to a closure. The duplicated request is unfortunate, but perhaps we can get rid of that later

Reply:
Yeah, I agree that it’s unfortunate but at least calling a pre-check on already pre-checked expression is effectively no-op...